helm/terraform-bucket-provision/templates/terraform-apply.yaml

```
@@ -55,6 +55,7 @@ spec:
               cd working;
               export TF_VAR_kubernetes_token=$( cat /var/run/secrets/kubernetes.io/serviceaccount/token );
               terraform init -backend-config=/etc/tf/gcs.tfbackend;
+              terraform workspace select -or-create {{ .Values.terraform.workspace }};
```

...Should we default the workspace to `.Release.Name` ? This way by default we don't override other deployment's terraform setup

```suggestion
              terraform workspace select -or-create {{ .Values.terraform.workspace | default .Release.Name }};
```

---

If we want to do that, we'd have to make sure any of the already-migrated repos have "default" in the workspace, otherwise, we'd need to re-migrate them all to `.Release.Name`. When Terraform first creates a State _without_ a Workspace, it just names it 'default' (even though it's not separated it can still target it with that). I thought if we stuck with that, it would simplify things if we didn't need to use workspaces for particular repos.

---

Ah I see. Sounds good! Let's move that to a tech debt ticket then, I'll approve this great PR.